monkey/monkey_island/cc/repository/mongo_machine_repository.py

```diff
@@ -38,12 +38,6 @@ def upsert_machine(self, machine: Machine):
         except Exception as err:
             raise StorageError(f'Error updating machine with ID "{machine.id}": {err}')
 
-        if result.matched_count != 0 and result.modified_count != 1:
-            raise StorageError(
-                f'Error updating machine with ID "{machine.id}": Expected to update 1 machine, '
-                f"but {result.modified_count} were updated"
-            )
-
```

Comment:
> It doesn't make sense to raise an error if upsert did no changes, because the purpose of "upsert" method is to ensure that data is up-to-date. If no changes were made it means it's already up-to-date.

Should upsert have even been called then? 
Let's at least add a log statement regarding this. This seems like useful debugging information if we run into bugs.

Reply:
Yes, it should've been called then, because the alternative is double the amount of queries (you check if DB is up to date and then update it) for the same result(DB is up to date).

Reply:
We would need to log everything then. Upserting when the machine is already up to date seems like the most common and least valuable information. So far I think whatever receives the error should be logging it, because that's the information we care about

Reply:
Fair

Reply:
I'm also not sure when would this case apear? when we accidentaly upsert more machines than aticipated?

Reply:
It would happen if there's some weird mongo error that didn't raise an exception :shrug:. I can't tell from the documentation if it's possible, so it's safer for us to handle it.

Reply:
This is what I get though:
![image](https://user-images.githubusercontent.com/36815064/194892531-4ff34e7d-e383-4da3-9280-732913ea009f.png)

Reply:
That's the only thing I've found in the docs: 
![image](https://user-images.githubusercontent.com/36815064/194893460-b4231fc2-851f-49ba-8fd2-ad1ae4de0582.png)

Reply:
That seems to indicate that my assumption based on the documentation was wrong.

It looks like the best thing to do drop https://github.com/guardicore/monkey/pull/2409/commits/d71791750a038d3793cb6fe12df6478802649c82

Reply:
On `develop`, I added a unit test for the same condition in `MongoAgentRepository` and removed this check. See the commit message on https://github.com/guardicore/monkey/commit/7fc0aaf42b80d592430d8bc9263ee61777095dca and the diff for more details.

Reply:
> though I'm not sure why I was getting the 0's
See https://www.mongodb.com/docs/manual/reference/method/db.collection.replaceOne/#replace-with-upsert. Maybe there's something different in what you're upserting, ie, it's a new document.

Reply:
[This](https://stackoverflow.com/a/39714522) may have something to do with it.

Reply:
[This](https://stackoverflow.com/a/39714522) may be useful.